src/core/p5.Renderer2D.js

```diff
@@ -392,6 +392,40 @@ p5.Renderer2D.prototype.updatePixels = function(x, y, w, h) {
       pixelsState.imageData;
   }
 
+  let pixels = pixelsState.pixels;
+  if (this._pInst instanceof p5 && this._pInst._isGlobal === true) {
+    pixels = globalThis.pixels;
+  }
+
+  if (pixels !== pixelsState.imageData.data) {
+    let imgData;
+    if (pixels instanceof Uint8ClampedArray) {
+      if (pixels.length === pixelsState.imageData.data.length) {
+        imgData = new ImageData(pixels, pixelsState.width, pixelsState.height);
+      } else {
+        let imgArr = Array.from(pixels);
```

Comment:
I wonder if it would be more efficient to use the TypedArray's `set()` method to get the array to the right size, like the one shown [here](https://stackoverflow.com/a/40324738). So instead of creating an array and a Uint8ClampedArray, we just create one Uint8ClampedArray of the right size and copy the existing data over.